src/analyses/variable-sensitivity/abstract_environment.cpp

```diff
@@ -53,59 +54,88 @@ std::vector<abstract_object_pointert> eval_operands(
   const abstract_environmentt &env,
   const namespacet &ns);
 
+bool is_ptr_diff(const exprt &expr)
+{
+  return (expr.id() == ID_minus) &&
+         (expr.operands()[0].type().id() == ID_pointer) &&
+         (expr.operands()[1].type().id() == ID_pointer);
+}
+
+bool is_ptr_comparison(const exprt &expr)
+{
+  auto const &id = expr.id();
+  if(id == ID_not)
```

Comment:
Are you handling `ID_and` elsewhere?  If so, should you handle `ID_not` there rather than here?

Reply:
Yep, sorted